### ISSUE TITLE
Add a toggle for double/single click in settings

### ISSUE THREAD

**Issue by KarkanAlzwayed** [elementary/files#1319]
A toggle in settings to enable/disable clicking/single click on files would be a great option that would work for a ton of people. I personally love the single click style, but I am probably one in a million, but it would definitely end a never ending discussion on the subreddit
Thank you

**Comment by CainCosta** [comment]
An comment on this,
I understand the hesitation to add a setting menu to Files itself.

Perhaps this could be handled in System Settings > Mouse & Touchpad > Clicking ?

**Comment by jeremypw** [comment]
@KarkanAlzwayed  To clarify - do you mean a toggle in the accessible in the Files window through a menu or button?  There is already a setting to toggle this at `io.elementary.files.preferences.single-click` which can be changed using dconf-editor or the commandline `gsettings set io.elementary.files.preferences single-click false`

**Comment by jeremypw** [comment]
I did draft a FIles settings plug for Switchboard several years ago but the idea of putting application settings in the System Settings was not favoured.

**Comment by KarkanAlzwayed** [comment]
> @KarkanAlzwayed  To clarify - do you mean a toggle in the accessible in the Files window through a menu or button?  There is already a setting to toggle this at `io.elementary.files.preferences.single-click` which can be changed using dconf-editor or the commandline `gsettings set io.elementary.files.preferences single-click false`

I don't understand your question. But now, there is a toggle in settings/tweaks/files that enables and disable double click. Why can't this be in the system without installing tweaks. Why do I have to go dig in dconf editor?

**Comment by jeremypw** [comment]
I think the answer in case of Files is that this app is derived from a pre-existing app (Marlin) and that was based on code from other filemanagers.  The decision was taken by elementary design team to make "single-click" the default option and also, in line with the elementary design philosophy, to simplify the UI as much as possible. But the setting has never been removed (although that has been mooted - see #222). Elementary is, as the name suggests, aimed more at non-technical users who do not want a lot of confusing options and are willing to accept the elementary way of doing things.  I appreciate that is not going to please everyone.

I explained why I thought @CainCosta 's idea would probably not be acceptable in my previous comment.

**Comment by jeremypw** [comment]
I do accept that some elementary native apps do expose some settings in the UI and there is a branch that adds a click-mode menu: https://github.com/elementary/files/tree/click-mode-menu

**Comment by jeremypw** [comment]
I have pushed a new minimal implementation of a UI for this setting for comment.

**Comment by m93a** [comment]
I'm trying to help my family (previously Windows users) move to elementary os and they find mixed mode weird and discouraging. In elementary 5, I could use the following command to re-enable double click:
```
gsettings set io.elementary.files.preferences single-click false
```
However, this option [seems to have disappeared](https://elementaryos.stackexchange.com/q/28079/19079) in elementary 6. Will there still be any way to re-enable double click in Odin?

**Comment by danirabbit** [comment]
This was fixed in 6.2.0!
